packages/ckeditor5-dev-env/lib/release-tools/tasks/bumpversions.js

```diff
@@ -0,0 +1,326 @@
+		for ( const packagePath of pathsCollection.matched ) {
+			const packageJson = getPackageJson( packagePath );
+
+			if ( !dependencies.has( packageJson.name ) ) {
+				dependencies.set( packageJson.name, packageJson.version );
+			}
+		}
+
+		return { packages, dependencies };
+	}
+
+	// Filter out packages which won't be released.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @params {Map} dependencies
+	// @returns {Object}
+	function filterPackagesThatWillNotBeReleased( packages, dependencies ) {
+		logDryRun( 'Filtering out packages that will not be updated...' );
+
+		for ( const pathToPackage of pathsCollection.matched ) {
+			const packageName = getPackageJson( pathToPackage ).name;
+
+			if ( !packages.has( packageName ) ) {
+				pathsCollection.matched.delete( pathToPackage );
+			}
+		}
+
+		return { packages, dependencies };
+	}
+
+	// Update versions of all dependencies.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @params {Map} dependencies
+	// @returns {Promise.<Map.<String, ReleaseDetails>>}
+	function updateDependenciesOfPackages( packages, dependencies ) {
+		logProcess( 'Updating dependencies for packages that will be released...' );
+
+		return executeOnPackages( pathsCollection.matched, repositoryPath => {
+			process.chdir( repositoryPath );
+
+			const packageJson = getPackageJson( repositoryPath );
+
+			log.info( `\nUpdating dependencies for "${ chalk.underline( packageJson.name ) }"...` );
+
+			updateDependenciesVersions( dependencies, path.join( repositoryPath, 'package.json' ) );
+
+			if ( exec( 'git diff --name-only package.json' ).trim().length ) {
+				if ( dryRun ) {
+					logDryRun( 'These changes would be committed.' );
+
+					log.info( chalk.grey( exec( 'git diff --word-diff package.json' ) ) );
+					exec( 'git checkout package.json' );
+				} else {
+					exec( 'git add package.json' );
+					exec( 'git commit -m "Internal: Updated dependencies. [skip ci]"' );
+				}
+			}
+
+			return Promise.resolve();
+		} ).then( () => packages );
+	}
+
+	// Gather descriptions of the release for all packages.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @returns {Promise.<Map.<String, ReleaseDetails>>}
+	function getLatestChangesForPackagesThatWillBeReleased( packages ) {
+		logProcess( 'Gathering changes from changelog for specified versions...' );
+
+		return executeOnPackages( pathsCollection.matched, repositoryPath => {
+			process.chdir( repositoryPath );
+
+			const packageJson = getPackageJson( repositoryPath );
+			const releaseDetails = packages.get( packageJson.name );
+
+			releaseDetails.changes = getChangesForVersion( releaseDetails.version, repositoryPath );
+
+			return Promise.resolve();
+		} ).then( () => packages );
+	}
+
+	// Validate the repositories.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @returns {Promise.<Map.<String, ReleaseDetails>>}
+	function validateRepositories( packages ) {
```

Comment:
this function can be rewritten to be synchronous.

Reply:
A few functions in this code could be synchronous. I made these asynchronous because if we will add some asynchronous actions to do, a body (core) of these functions will not require a rewrite.

They are asynchronous as well because I wanted to show my intention: "execute given action on all packages". This mechanism occurs in the whole script.